src/Controller/WebPageArchiveController.php

```diff
@@ -53,4 +57,48 @@ public function title($web_page_archive) {
     return $web_page_archive->label();
   }
 
+  /**
+   * Common batch processing callback for all operations.
+   */
+  public static function batchProcess(WebPageArchive $web_page_archive, &$context) {
+    $queue = \Drupal::service('queue')->get("web_page_archive_capture.{$web_page_archive->uuid()}");
+    $queue_worker = \Drupal::service('plugin.manager.queue_worker')->createInstance('web_page_archive_capture');
+
+    // TODO: Move threshold into admin panel.
+    // Per-entity or global for all settings?
+    $number_of_queue = ($queue->numberOfItems() < WEB_PAGE_ARCHIVE_BATCH_SIZE) ? $queue->numberOfItems() : WEB_PAGE_ARCHIVE_BATCH_SIZE;
+
+    for ($i = 0; $i < $number_of_queue; $i++) {
+      // Claim item and attempt to process it.
+      if ($item = $queue->claimItem()) {
```

Comment:
Just so we have a record of it on GitHub... We talked about this on slack, and decided that it makes the most sense to only batch a single item at a time.  

![image](https://user-images.githubusercontent.com/5263371/28078342-24d6f3f6-662a-11e7-9f0d-3cb447f73fc2.png)

It gives you a clear understanding of how many operations need to complete, and reduces risk of timeouts. And while it might actually be a little slower, it feels faster, simply because you get quicker feedback in the UI. 

This fix is included in #7, not this PR.